cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -1358,42 +1364,76 @@ private int moveLeadershipInBatch() throws InterruptedException, ExecutionExcept
             }
           }
         }
         sendSlowExecutionAlert(slowTasksToReport);
-        handleDeadOrAbortingReplicaActions(deadOrAbortingInterBrokerReplicaActions);
+        handleDeadInterBrokerReplicaTasks(deadInterBrokerReplicaTasks, stoppedInterBrokerReplicaTasks);
         updateOngoingExecutionState();
       } while (!inExecutionTasks().isEmpty() && finishedTasks.isEmpty());
 
       LOG.info("Finished tasks: {}.{}{}{}", finishedTasks,
-               forceStoppedTaskIds.isEmpty() ? "" : String.format(". [Force-stopped: %s]", forceStoppedTaskIds),
+               stoppedTaskIds.isEmpty() ? "" : String.format(". [Stopped: %s]", stoppedTaskIds),
                deletedTaskIds.isEmpty() ? "" : String.format(". [Deleted: %s]", deletedTaskIds),
-               deadOrAbortingTaskIds.isEmpty() ? "" : String.format(". [Dead/aborting: %s]", deadOrAbortingTaskIds));
+               deadTaskIds.isEmpty() ? "" : String.format(". [Dead: %s]", deadTaskIds));
 
       return finishedTasks;
     }
 
-    private void handleDeadOrAbortingReplicaActions(List<ExecutionTask> deadOrAbortingInterBrokerReplicaActions)
+    /**
+     * Attempts to cancel/rollback the ongoing reassignment of dead/stopped inter-broker replica actions and stops the
+     * execution if not already requested so by the user.
+     *
+     * If all dead tasks are due to stopped inter-broker replica tasks, it waits until the rollback is completed.
+     * Otherwise, it will not wait for the actual rollback to complete to avoid being blocked on a potentially stuck
+     * reassignment operation due to dead brokers in the cluster. If by the time the next execution is attempted, the
+     * rollback is still in progress on Kafka server-side, the executor will detect the ongoing server-side execution
+     * and will not start a new execution (see {@link #sanityCheckOngoingMovement}).
+     *
+     * @param deadInterBrokerReplicaTasks Inter-broker replica tasks that are marked as dead due to dead destination brokers.
+     * @param stoppedInterBrokerReplicaTasks Inter-broker replica tasks that are marked as dead due to being stopped by user.
+     */
+    private void handleDeadInterBrokerReplicaTasks(List<ExecutionTask> deadInterBrokerReplicaTasks,
+                                                   List<ExecutionTask> stoppedInterBrokerReplicaTasks)
         throws InterruptedException, ExecutionException, TimeoutException {
-      if (!deadOrAbortingInterBrokerReplicaActions.isEmpty()) {
-        ExecutionUtils.submitReplicaReassignmentTasks(_adminClient, deadOrAbortingInterBrokerReplicaActions);
+      List<ExecutionTask> tasksToCancel = new ArrayList<>(deadInterBrokerReplicaTasks);
+      tasksToCancel.addAll(stoppedInterBrokerReplicaTasks);
+
+      if (!tasksToCancel.isEmpty()) {
+        // Cancel/rollback reassignment of dead inter-broker replica tasks.
+        ExecutionUtils.submitReplicaReassignmentTasks(_adminClient, tasksToCancel);
```

Comment:
I am confused about the naming at the very beginning since the comment says "Cancel/rollback ... tasks" but we call `submitReplicaReassignmentTasks` here. When I look at the [implementation ](https://github.com/linkedin/cruise-control/blob/6b7157f8b4b5f77394d4577c3f3c68d623edfe45/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionUtils.java#L235) I realized that the task will be cancelled if it is in a DEAD state. Are the tasks in `deadInterBrokerReplicaTasks` and `stoppedInterBrokerReplicaTasks` all guaranteed to be in DEAD state?

Maybe add more documentations here to make it clearer?

Reply:
I added a sanity check along with `expected to be ...` phrase in the existing `param` in the JavaDoc.